tests/certs_test.go

```diff
@@ -1,28 +1,422 @@
+			nonExistentCertName := "non-existent-cert"
+
+			addDomain(domain, testApp.Name)
+
+			// attempt to add cert with improper cert name (includes periods)
+			sess, err := start("deis certs:add %s %s %s", domain, certs["wildcard"].CertPath, certs["wildcard"].KeyPath)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("400 Bad Request"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to add cert with cert and key file swapped
+			sess, err = start("deis certs:add %s %s %s", certName, certs["wildcard"].KeyPath, certs["wildcard"].CertPath)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("400 Bad Request"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to add cert with non-existent keys
+			sess, err = start("deis certs:add %s %s %s", certName, nonExistentCert, "non-existent.key")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("open %s: no such file or directory", nonExistentCert))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to remove non-existent cert
+			sess, err = start("deis certs:remove %s", nonExistentCertName)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to get info on non-existent cert
+			sess, err = start("deis certs:info %s", nonExistentCertName)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to attach non-existent cert
+			sess, err = start("deis certs:attach %s %s", nonExistentCertName, domain)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to detach non-existent cert
+			sess, err = start("deis certs:detach %s %s", nonExistentCertName, domain)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			addCert(certName, certs["wildcard"].CertPath, certs["wildcard"].KeyPath)
+
+			// attempt to attach to non-existent domain
+			sess, err = start("deis certs:attach %s %s", certName, "non-existent-domain")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to detach from non-existent domain
+			sess, err = start("deis certs:detach %s %s", certName, "non-existent-domain")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			// attempt to remove non-existent cert
+			sess, err = start("deis certs:remove %s", nonExistentCertName)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("404 Not Found"))
+			Eventually(sess).Should(Exit(1))
+
+			removeCerts([]string{certName})
+		})
+
+		It("can add, attach, list, and remove certs: 2 domains, 1 cert", func() {
```

Comment:
Similar to question above, Do we believe this scenario is still needed since we perform similar actions [here?] (https://github.com/deis/workflow-e2e/pull/75/files#diff-4e69408eafdc533cb4f2550813c5ba3cR331) (verifying SSL endpoint on domain w/ cert attached, verifying lack of SSL endpoint on domain(s) w/o cert(s) attached)